### ISSUE TITLE
Phone / Lofi effect

### ISSUE THREAD

**Issue by topisani** [topisani/OTTO#26]
We need dsp and graphics for a lofi/phone/radio type of effect.

**Comment by D-I-S-K-U-S** [comment]
I will see if i can come up with a cool idea for the graphics. :)

**Comment by jmidt** [comment]
So, since i can't find a good rundown of the effect on video: What is it? It seems like mostly some bitcrushing - perhaps with some granular elements. There's definitely some glitchyness going on...

**Comment by topisani** [comment]
the OP-1 has an effect that does i think a bit of analogue sounding distortion + definately some bit crushing and some "glitchyness", which seems a bit like a randomly/envelope triggered gate. But i dont think this is necessarilly what we have to do with it. I just think we need something to destroy sound, both analogue-like and digital (bit crushing)

**Comment by D-I-S-K-U-S** [comment]
@jmidt [Short video of '_Phone_' effect](https://youtu.be/DN6zXJTtaNw?t=200) _(starts at 3:20 if timestamp doesn't work)_.

**Comment by Hate9** [comment]
I don't know about the OP-1 effect, but in general, the reason phones sound the way they do is because of heavy signal filtering (e.g. hard low and high passing), so maybe it could be a combination of hard passing, noise modulation, and bitcrushing.

**Comment by D-I-S-K-U-S** [comment]
_Current state of this effect's UI._
<img width="955" alt="schermafbeelding 2017-12-02 om 00 21 40" src="https://user-images.githubusercontent.com/31598745/33524685-9ddefbc2-d821-11e7-8b00-fb2015ee57d1.png">

**Comment by nebularnoise** [comment]
@D-I-S-K-U-S That OP1 effect sounds like Linear Predictive Coding, as used in the Speak&Spell, or Bitspeek for instance. I have some material on that, I'll try and see what I can dig up.

As for the 'traditional' lofi phone sound, yeah, it's mostly highpassing at 300Hz and downsampling at 7kHz. Then, predistortion and bitcrushing can be added for good measure.

**Comment by nebularnoise** [comment]
LPC is far from trivial for me to implement from scratch, but there are a couple of open source DSP libraries that do it. We could either import it, or inspire from their code.

- [Soundpipe lpc module](http://paulbatchelor.github.io/res/soundpipe/docs/lpc.html) ([repo](https://github.com/PaulBatchelor/Soundpipe))
- [SNDTools lpc](https://github.com/ccrma/sndtools/blob/master/src/rt_lpc/lpc.cpp)
- [VoiceCodec](https://github.com/longluo/VoiceCodec/tree/master/src)

Soundpipe looks the most interesting IMHO. The only downside is that Soundpipe's lpc is limited to a 10 pole filter lpc. Don't get me wrong, 10 pole is exactly the kinda sound we're looking for, since it's the rough model used in the S&S, but adjustable filter order could have been nice to have.

Either way, if anyone's interested in the maths that are involved, I found [those slides](https://www.ece.ucsb.edu/Faculty/Rabiner/ece259/digital%20speech%20processing%20course/lectures_new/Lecture%2013_winter_2012_6tp.pdf) to be reasonably exhaustive.

**Comment by topisani** [comment]
status on this? For me to really be enthusiastic about it, it would need to be not too similar to the OP-1 phone.

Also, @D-I-S-K-U-S We need to rework that UI for the new style. To be honest, I'm not too sure about it anyway, but lets see whats possible in a rework

**Comment by D-I-S-K-U-S** [comment]
Yeah, we/I should probably come up with a new UI concept for this thats a bit more abstract. On it!

**Comment by D-I-S-K-U-S** [comment]
I was just looking through the Faust documentation, and I found there's vocal synthesis. 
Would that be anything we can work with @nebularnoise? 
A similar effect at least, not necessarily the same.

**Comment by jmidt** [comment]
Updated. Added **GFI** flair.

We need to finish this effect. Most of the DSP has already been written, it just needs to be wrapped up in a proper effects engine struct. The code currently in this PR: https://github.com/topisani/OTTO/pull/104 but should be moved to a proper feature/ branch.

Apart from the code, we also need new UI ideas in the new style.

**Comment by jmidt** [comment]
I have started work on this again. It is now in the feature/LPC branch
